MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -1629,8 +1629,9 @@ static void signalAndUntrack(const MVKSwapchainSignaler& signaler) {
 	// Ensure this image, the drawable, and the present fence are not destroyed while
 	// awaiting MTLCommandBuffer completion. We retain the drawable separately because
 	// a new drawable might be acquired by this image by then.
-	// Signal the fence from this callback, because the last one or two presentation
-	// completion callbacks can occasionally stall.
+	// Signal the fence and notify the swapchain that the present has completed
+	// from this callback, because the last one or two presentation completion
+	// callbacks can occasionally stall.
```

Comment:
Nice. Great catch!

I really struggled with these lost/stalled presentation callbacks at the time I added this. Thanks for figuring out some further requirements!